src/content/self-hosted/administration/configuration.mdx

```diff
@@ -623,6 +483,137 @@ registry:
         size: 40Gi
 ```
 
+##### azure
+
+Use this to use Azure Storage to store your private images.
+
+```yaml
+registry:
+  storage:
+    provider:
+      azure:
+        enabled: true
+        storage:
+          container: "storage-container-name"
+          accountName: "service-account-name"
+```
+
+Then, create a secret like this:
+
+```console
+$ kubectl create secret generic okteto-registry-secret --namespace=okteto --from-literal=secretKey=STORAGE_ACCOUNT_ACCESS_KEY
+```
+
+- `STORAGE_ACCOUNT_ACCESS_KEY`: A storage account access key with read/write permissions to the storage container specified in the configuration.
+
+
+##### aws
+
+Use this to use S3 to store your private images.
+
+###### IAM User
+
+```yaml
+registry:
+  storage:
+    provider:
+      aws:
+        enabled: true
+        bucket: "bucket-name"
+        region: "aws-region"
+        iam:
+          enabled: true
+          accessKeyID: "XXXXXX"
+```
+
+Then, create a secret like this:
+
+```console
+$ kubectl create secret generic okteto-registry-secret --namespace=okteto --from-literal=secretKey=IAM_SECRET_KEY
+```
+
+- `IAM_SECRET_KEY`: The secret key of an IAM with read/write permissions to the bucket specified in the configuration.
+
+###### IAM Role for Service Account (IRSA)
+
+```yaml
+registry:
+  storage:
+    provider:
+      aws:
+        enabled: true
+        bucket: "bucket-name"
+        region: "aws-region"
+        iam:
+          enabled: false
+  serviceAccount:
+    annotations:
+      eks.amazonaws.com/role-arn: 'arn:aws:iam::111122223333:role/okteto-registry-role-name'
+```
+
+##### digitalocean
+
+Use this to use DigitalOcean Spaces to store your private images.
+
+```yaml
+registry:
+  storage:
+    provider:
+      digitalocean:
+        enabled: true
+        space:
+          name: "space-name"
+          accessKeyID: "access-key"
+```
+
+Then, create a secret like this:
+
+```console
+$ kubectl create secret generic okteto-registry-secret --namespace=okteto --from-literal=secretKey=ACCESS_KEY
+```
+
+- `ACCESS_KEY`: The access key of the storage space specified in the configuration.
+
+##### gcp
+
+Use this to use Google Cloud Storage to store your private images.
+
+###### Service Account Key (JSON)
+
+```yaml
+registry:
+  storage:
+    gcp:
+      enabled: true
+      bucket: "bucket-name"
+      project: "project-id"
+```
+
+Then, create a secret like this:
+
+```console
+$ kubectl create secret generic okteto-registry-secret --namespace=okteto --from-file=secretKey=key.json
+```
+
+- `key.json`: A file containing the key of a service account with read/write permissions to the storage bucket specified in the configuration.
+
+
+###### Workload Identity
+
```

Comment:
as for AWS, we should add a description for each, and put this one first since it's the recommended day. Do we also need to link to GCP docs on this topic? (e.g. to enable it in the cluster)